Cython/Shadow.pyi

```diff
@@ -267,6 +267,53 @@ class typedef(CythonType, Generic[_T]):
 
 NULL: pointer[Any]
 
+# Lock types for synchronization
+class pymutex:
+    """
+    A low-cost mutex lock type.
+    In Python 3.13+, it uses the fast PyMutex implementation. In older Python versions,
+    it falls back to the heavier "PyThread_type_lock".
+    Can be used in 'with' statements and supports nogil context.
+    """
+    def acquire(self) -> None: ...
+    def release(self) -> None: ...
+    def locked(self) -> bool:
+        """
+        Check if the lock is currently held.
+        Returns True if locked, False otherwise.
+        Only available on Python 3.13+.
+        """
+        ...
+    def __enter__(self) -> None: ...
+    def __exit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc: Optional[BaseException],
+        tb: Optional[TracebackType]
+    ) -> None: ...
+
+class pythread_type_lock:
+    """
+    A mutex lock type using PyThread_type_lock.
+    Can be used in 'with' statements and supports nogil context.
+    Compatible with the Limited API.
+    """
+    def acquire(self) -> None: ...
+    def release(self) -> None: ...
+    def locked(self) -> bool:
+        """
+        Check if the lock is currently held.
+        Raises NotImplementedError as PyThread_type_lock doesn't support this operation.
```

Comment:
```suggestion
        Aborts with a fatal Python error as PyThread_type_lock does not support this operation.
```
There might be a way to express this in `typing`, although that could turn out to be unhelpful.